airflow/contrib/hooks/telegram_hook.py

```diff
@@ -0,0 +1,88 @@
+import telegram
+from time import sleep
+from airflow.hooks.base_hook import BaseHook
+from airflow.exceptions import AirflowException
+
+
+def telegram_retry(func, *args, **kwargs):
+    max_retries = kwargs.pop("max_retries", 5)
+    retry_sleep = kwargs.pop("retry_sleep", 0)
+    # perform request with retry
+    for retry in range(max_retries):
+        try:
+            return func(*args, **kwargs)
+        except (telegram.error.TimedOut,) as e:
+            if retry_sleep:
+                sleep(retry_sleep)
+
+
+class TelegramHook(BaseHook):
+    """
+       Interact with Telegram, using python-telegram-bot library.
+    """
+
+    def __init__(self, token=None, telegram_conn_id=None, chat_id=None):
+        """
+        Takes both telegram bot API token directly and connection that has telegram bot API token.
+        If both supplied, telegram API token will be used.
+        :param token: telegram API token
+        :type token: str
+        :param telegram_conn_id: connection that has telegram API token in the password field
+        :type telegram_conn_id: str
+        :param chat_id: Telegram public or private channel id (optional).
+        Check https://stackoverflow.com/a/33862907 to see how you can obtain chat_id for private
+        channel
+        :type chat_id: str
+        """
+        self.token = self.__get_token(token, telegram_conn_id)
+        self.chat_id = self.__get_chat_id(chat_id, telegram_conn_id)
+
+    def __get_token(self, token, telegram_conn_id):
+        if token is not None:
+            return token
+        elif telegram_conn_id is not None:
+            conn = self.get_connection(telegram_conn_id)
+
+            if not getattr(conn, "password", None):
```

Comment:
```suggestion
            if not conn.password:
```
The connection always contains a password field. Sometimes it is only empty.

Reply:
I've used SlackHook/SlackOperator from core as a reference implementation. Should we change  it too?

Reply:
You can do it in a separate PR. This isn't a big issues. The current implementation works correctly, but it can be written better.